crucible-llvm/src/Lang/Crucible/LLVM/MemModel.hs

```diff
@@ -1287,6 +1290,40 @@ loadRaw sym mem ptr valType alignment = do
   let gsym = unsymbol <$> isGlobalPointer (memImplSymbolMap mem) ptr
   G.readMem sym PtrWidth gsym ptr valType alignment (memImplHeap mem)
 
+-- | Load an array with concrete size from memory.
+loadArrayConcreteSizeRaw ::
+  forall sym wptr .
+  (IsSymInterface sym, HasPtrWidth wptr, Partial.HasLLVMAnn sym, ?memOpts :: MemOptions) =>
+  sym ->
+  MemImpl sym ->
+  LLVMPtr sym wptr ->
+  Natural ->
+  Alignment ->
+  IO (Either (Pred sym) (Pred sym, SymArray sym (SingleCtx (BaseBVType wptr)) (BaseBVType 8)))
```

Comment:
What is the relationship between the `sz :: Natural` argument and the `SymArray` that is returned? I think this is set up so that the elements of the `SymArray` in the range `[0, sz)` correspond to the elements of the array stored in memory as bytes, and all other elements are set to byte `0`. Worth stating this somewhere in the Haddocks.